src/app.py

```diff
@@ -697,6 +720,127 @@ def render_form(**kwargs):
                        edit_url=edit_url, site_dir_url=site_dir_url,
                        new_url=new_url)
 
+
+@app.route("/generate_project_yaml", methods=["GET", "POST"])
+def generate_project_yaml():
+
+    def render_form(**kwargs):
+        institutions = list(global_data.get_mappings().project_institution.items())
+        session.pop("form_data", None)
+
+        return render_template("generate_project_yaml.html.j2", form=form, infos=form.infos, institutions=institutions, **kwargs)
+
+    def validate_project_name(form, field):
+        project_names = set(x['Name'] for x in global_data.get_projects()['Projects']['Project'])
+        if field.data in project_names:
+            raise ValidationError(f"{field.data} is already registered in OSG Topology. ")
+
+    form = GenerateProjectForm(request.form, **request.args, **session.get("form_data", {}))
+    form.field_of_science.choices = _make_choices(global_data.get_mappings().nsfscience.keys(), select_one=True)
+
+    # Add this validator if it is not their
+    if not len(form.project_name.validators) > 1:
+        form.project_name.validators.append(validate_project_name)
+
+    # If they have returned after logging into Github make the Submission button stand out
+    if "github_login" in session:
+        form.auto_submit.label.text = "Submit Automatically"
+        form.auto_submit.render_kw = {
+            "class": "btn btn-warning"
+        }
+
+    # Report any external errors
+    if "error" in request.args:
+        return render_form(error=request.args['error'])
+
+    # Anything past this point needs a valid form
+    if not form.validate_on_submit():
+        return render_form()
+
+    # If the user has their Github verified
+    if request.method == "POST" and "github_login" in session and "auto_submit" in request.form:
+
+        try:
+            # Gather necessary data
+            create_pr_data = {
+                "file_path": f"data/{request.values['project_name']}.yaml",
+                "file_content": form.get_yaml(),
+                "branch": f"add-project-{request.values['project_name']}",
+                "message": f"Add Project {request.values['project_name']}",
+                "committer": GithubUser.from_token(session["github_login"]['access_token']),
+                "fork_repo": GitHubRepoAPI(GitHubAuth("osg-doc-bot", app.config["OSG_BOT_GITHUB_TOKEN"]), 'osg-doc-bot', 'topology'),
+                "root_repo": GitHubRepoAPI(GitHubAuth("osg-doc-bot", app.config["OSG_BOT_GITHUB_TOKEN"]), 'opensciencegrid', 'topology'),
```

Comment:
So for what it's worth i had intended the GitHubAuth/GitHubRepoAPI setup to look a bit more like this:

```suggestion
                "root_repo": GitHubAuth(self.form_gh_api_user, self.form_gh_api_token, app.logger).target_repo('opensciencegrid', 'topology'),
```

which returns the GitHubRepoAPI object ... I intended never to have to _type_ GitHubRepoAPI.

(Oh and the extra `app.logger` arg i added to GitHubAuth is optional.)